test/functional/test_framework/test_node.py

```diff
@@ -140,6 +149,23 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             AddressKeyPair('mzRe8QZMfGi58KyWCse2exxEFry2sfF2Y7', 'cPiRWE8KMjTRxH1MWkPerhfoHFn5iHPWVK5aPqjW8NxmdwenFinJ'),
     ]
 
+    def get_valgrind_version(self):
+        """Returns valgrind version (major, minor) in a tuple"""
+        try:
+          rc = subprocess.call(['which', 'valgrind'],
```

Comment:
Use `shutil.which("valgrind")` to avoid shelling out unnecessarily :)

Reply:
Yup, much better!